sdcflows/utils/wrangler.py

```diff
@@ -244,46 +256,68 @@ def find_estimators(*, layout, subject, sessions=None, fmapless=True, force_fmap
 
     """
     from .. import fieldmaps as fm
+    from .misc import create_logger
     from bids.layout import Query
     from bids.exceptions import BIDSEntityError
 
+    # Set up logger (logs only visable if verbose)
+    logger = create_logger('sdcflows.wrangler', level=10 if verbose else 40)
+
     base_entities = {
         "subject": subject,
         "extension": [".nii", ".nii.gz"],
         "scope": "raw",  # Ensure derivatives are not captured
     }
 
     subject_root = Path(layout.root) / f"sub-{subject}"
-    sessions = sessions if sessions else layout.get_sessions(subject=subject)
-
+    sessions = sessions or layout.get_sessions(subject=subject)
     fmapless = fmapless or {}
     if fmapless is True:
         fmapless = {"bold", "dwi"}
 
     estimators = []
 
-    b0_ids = tuple()
+    # Step 1. Use B0FieldIdentifier metadata
+    has_B0FI = False
     with suppress(BIDSEntityError):
         b0_ids = layout.get_B0FieldIdentifiers(**base_entities)
+        has_B0FI = True
+
+    if has_B0FI:
+        logger.debug(
+            "Dataset includes `B0FieldIdentifier` metadata."
+            "Any data missing this metadata will be ignored."
+        )
+        for b0_id in b0_ids:
+            # Found B0FieldIdentifier metadata entries
+            b0_entities = base_entities.copy()
+            b0_entities["B0FieldIdentifier"] = b0_id
+
+            e = fm.FieldmapEstimation([
+                fm.FieldmapFile(fmap.path, metadata=fmap.get_metadata())
+                for fmap in layout.get(**b0_entities)
+            ])
+            logger.debug(
+                "Found %s estimation from %d sources:\n- %s",
+                e.method.name,
+                len(e.sources),
+                "\n- ".join([str(x.path.relative_to(layout.root)) for x in e.sources]),
+            )
```

Comment:
With as many times as you've written exactly this, I think it could be useful to add a function or method that would emit a description:

```Python
def describe(est, root):
    return "\n- ".join(
        [f"Found {est.method.name} estimation from {len(est.sources)} sources:"] +
        [str(src.path.relative_to(root)) for src in est.sources]
    )
```

Then all these debug statements become:

```suggestion
            logger.debug(describe(e, layout.root))
```
or

```suggestion
            logger.debug(e.describe(layout.root))
```